### ISSUE TITLE
Training taking TOO long with no idea of right or wrong

### ISSUE THREAD

**Issue by Anish1906** [matterport/Mask_RCNN#2806]
I have been trying to run the shapes.py code on various platforms. I use an M1 Pro mac, and I have tried using the Mac terminal, Colab GPU, and even the university cluster. But the code is stuck at the following:

Starting at epoch 0. LR=0.002

Checkpoint Path: /Users/anishphule/Desktop/M_RCNN/logs/shapes20220411T0355/mask_rcnn_shapes_{epoch:04d}.h5
Selecting layers to train
fpn_c5p5               (Conv2D)
fpn_c4p4               (Conv2D)
fpn_c3p3               (Conv2D)
fpn_c2p2               (Conv2D)
fpn_p5                 (Conv2D)
fpn_p2                 (Conv2D)
fpn_p3                 (Conv2D)
fpn_p4                 (Conv2D)
rpn_model              (Functional)
mrcnn_mask_conv1       (TimeDistributed)
mrcnn_mask_bn1         (TimeDistributed)
mrcnn_mask_conv2       (TimeDistributed)
mrcnn_mask_bn2         (TimeDistributed)
mrcnn_class_conv1      (TimeDistributed)
mrcnn_class_bn1        (TimeDistributed)
mrcnn_mask_conv3       (TimeDistributed)
mrcnn_mask_bn3         (TimeDistributed)
mrcnn_class_conv2      (TimeDistributed)
mrcnn_class_bn2        (TimeDistributed)
mrcnn_mask_conv4       (TimeDistributed)
mrcnn_mask_bn4         (TimeDistributed)
mrcnn_bbox_fc          (TimeDistributed)
mrcnn_mask_deconv      (TimeDistributed)
mrcnn_class_logits     (TimeDistributed)
mrcnn_mask             (TimeDistributed)
WARNING:tensorflow:multiprocessing can interact badly with TensorFlow, causing nondeterministic deadlocks. For high performance data pipelines tf.data is recommended.

It stays like this for hours and the log file doesn't change, niether does the epoch 1 appear. Someone please help!! I'm trying to use this for my term project!

**Comment by babanthierry94** [comment]
Try to reduce those parameters and try again. Usually is a problem of ressources availabillity
```
IMAGES_PER_GPU = 1
IMAGE_RESIZE_MODE = "square"
IMAGE_MIN_DIM = 512 
IMAGE_MAX_DIM = 512 # Small images take less memory
```

After that fill free to adjust the parameters to fit with your data.

**Comment by ErinYang96** [comment]
@babanthierry94 @Anish1906 Hi, do you solve the problem? I just have the same problem as you do by using [Mask-RCNN-TF2.7.0-keras2.8.0](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.8.0). I test on the server with NVIDIA RTX A5000 (24G), and have already set the parameters above as low as possible. I can't solve it still. Maybe you have some new ideas to solve it? I am glad to know if you do.
I post my questions [here](https://github.com/Kamlesh364/Mask-RCNN-TF2.7.0-keras2.8.0/issues/5)

**Comment by babanthierry94** [comment]
Maybe you have a problem of compatibility. 
I am using TF2.1 and keras 2.4.3. 
I used this Matterport fork project compatible with tensorflow 2.1. https://github.com/tomgross/Mask_RCNN/tree/71003370a447cbe39430622f007ea83bb1815c2b

**Comment by JPVercosa** [comment]
I dont know wich version of tensorflow, CUDA and CUDNN you are using, but I had a lot of problems on this part of the code because of compatibility of packages versions. I'm currently working in Windows, but if you fix that probably the model will work. 

The only way I made it around and trained my model, was creating a freshly new conda env, installing manually `pip install tensorflow==2.5.0`, then editing `requirements.txt` and removing 'tensorflow' from it, for it dont override it the one that I just installed and finally ran `pip install -r requirements.txt` (or pip3, depending on OS), to install dependencies, and 'python setup.py install' to install mask-rcnn package.

I'm doing a second thing, that make big difference, I'm currently working [with this fork from akTwelve](https://github.com/akTwelve/Mask_RCNN), that is updated to tensorflow >2.X. 

With TensorFlow 2.5.0, accordingly to this table of [tensorflow.org](https://www.tensorflow.org/install/source#tested_build_configurations), we should have also installed CUDA 11.2 and cuDNN 8.1:
![image](https://user-images.githubusercontent.com/27090431/190140645-9015201e-a0b8-469f-8953-a89b4b7b8c31.png)

I had a newer version of CUDA and cuDNN running in my PC and the code won't work. Only when I installed exactly those version of CUDA and cuDNN, and add their respective PATH to the system enviroment variables (mandatory step on Windows OS), with the correct version of TensorFlow, things started to work.
